lib/types.d.ts

```diff
@@ -0,0 +1,32 @@
+type MergeBy<TObject, TOtherObject> = Omit<TObject, keyof TOtherObject> & TOtherObject;
+
+type DeepRecord<TKey extends string | number | symbol, TValue> = {[key: string]: TValue | DeepRecord<TKey, TValue>};
+
+type DeepKeyOf<TObject> = TObject extends object
+    ? {
+          [TKey in keyof TObject & (string | number)]: TObject[TKey] extends Record<string, unknown> ? `${TKey}.${DeepKeyOf<TObject[TKey]>}` | TKey : TKey;
+      }[keyof TObject & (string | number)]
+    : TObject;
```

Comment:
Ouch this hurts my brain. Is this too complicated or am I just a beginner? Idk. If we're going to keep this let's please add a description AND some examples in a comment.

Reply:
@neil-marcellini Will add comments explaining.

Reply:
Why are we using both `object` and `Record<string, unknown>` for checking if a field value is object or not? Can we just use either of these?

Reply:
@fabioh8010 this is to get keys from nested keys (because we have nested keys other than collection?)

Reply:
@neil-marcellini @hayata-suenaga @blazejkustra Sorry actually this type is not being used anymore, will remove it.